src/common/resource/CResourceHolder.cpp

```diff
@@ -282,6 +282,28 @@ bool CResourceHolder::LoadResourceSection( CScript * pScript )
 //*********************************************************
 // Resource Section Definitions
 
+lpctstr CResourceHolder::ResourceGetName(const CResourceIDBase& rid, RES_TYPE iExpectedType)
+{
+    ADDTOCALLSTACK("CResourceHolder::ResourceGetNameFromMore");
+    CResourceID ridValid = CResourceID(iExpectedType, 0);
+    if (!rid.IsValidResource())
+    {
+        if (rid.GetResIndex() != 0)
+        {
+            g_Log.EventError("Expected a valid resource. Ignoring it/Converting it to an empty one.\n");
+        }
+    }
+    else if (rid.GetResType() != iExpectedType)
+    {
+        g_Log.EventWarn("Expected resource with type %d, got %d. Ignoring it/Converting it to an empty one.\n", iExpectedType, rid.GetResType());
+    }
+    else
+    {
+        ridValid = rid;
+    }
+    return g_Cfg.ResourceGetName(ridValid); // Even it's 0, we should return it's name, as it can be mr_nothing.
```

Comment:
Good catch.
Still i'd keep somewhere this check:
```
tatic lpctstr _ridGetName(CResourceID const& rid, const RES_TYPE iExpectedType)
{
    /*
    This RID (ResourceID) is read directly from the scripts, and it might contain a null/zero value (because it was blank or undefined, for example). Using a CResourceID with a zero internal value would generate a warning log message, because it's, actually, invalid (no resource type data, no resource internal flag), and the CResourceID class itself has some internal sanity checks.
    
    Acknowledge that there might exist the case where we want the data contained in rid to be a invalid CResourceID
    (e.g. a fully scripted item with custom behavior, in this case the MORE might not be used to contain a UID or a valid CResourceID but a simple number used as counter or whatever)
    */
    
    const CResourceDef* pResDef = g_Cfg.RegisteredResourceGetDef(rid);
    const auto* pBaseDef = dynamic_cast<CBaseBaseDef*>(pResDef);
    ASSERT(pBaseDef);
    
    // This item doesn't use an hardcoded behavior, so its MORE value might be used for custom stuff.
    if (pBaseDef->Can(CAN_I_SCRIPTEDMORE))
        return rid;
```
Otherwise we'll have people complaining of warnings while they are using MORE values for custom purposes, on scripted skills

Reply:
I changed the system a bit and added ResourceGetName function under CItem to check Can Flag, if you tell this change is fine, I can implement SCRIPTED_MORE flag in the next pr, also I can update changelog.

Reply:
Mmm why are you passing also the raw more value?

Reply:
Because if it's a SCRIPTEDMORE the given value could be just a numbers instead of resource. So I return it as DWORD if it's not a valid resource, but I can take a look in a minute if it causes any issue to getting dword from it's own struct.